### PR TITLE
Build moose_test-opt in the Conda MOOSE package

### LINKED ISSUE #26493: Build moose_test-opt (test application) in Conda MOOSE package
## Reason
It is useful to have the ability to test everything, and not only modules, with the Conda moose package.

## Design
Add `moose/test` to the build.sh script for `make install`.

## Impact
The ability to run tests in the tests directory using pre-built moose_test-opt

### PR DESCRIPTION
**Pull request by milljm**
Add `moose_test-opt` to the Conda MOOSE package.

Also using this PR to troubleshot the pesky `libwasp` library missing. There are actually many more libraries missing. But libWasp happens to be the first discovered.

Closes #26493

### CODE REVIEW

**Comment by loganharbour** [comment]
I don't think the issue you have tagged for the test skipping is valid; we should make an issue with the problem not the issue that's set to just make things work

**Comment by loganharbour** [comment]
many of these are already skipped and have a proper issue, looks like you just had an old moose:

```
  [json_full]
    type = 'PythonUnitTest'
    input = 'test_json.py'
    test_case = "TestFull"
    max_buffer_size = -1
    issues = '#7855 #7661 #2881 #10839 #12455'
    design = 'JsonInputFileFormatter.md MooseApp.md'
    installation_type = in_tree # see #26480
    requirement = 'The system shall be able to dump input file syntax in JSON format.'
    # Allow tests to pass when run out of tree #26493
    installation_type = in_tree
  []
```

https://github.com/idaholab/moose/blob/8286ffb88a987d3bacb7ad58b18620f2c4632541/test/tests/outputs/format/tests for example

**Comment by milljm** [comment]
> many of these are already skipped and have a proper issue, looks like you just had an old moose:

Yes, I figured i missed a few

**Comment by milljm** [comment]
> I don't think the issue you have tagged for the test skipping is valid; we should make an issue with the problem not the issue that's set to just make things work

I don't even know what the problem is. What issue shall I use?

**Comment by milljm** [comment]
Looks like you nailed all the skips I needed. I had a few extras. I was expecting those to fail, but they were not. Fortunate surprise.

**Comment by milljm** [comment]
This is finished. Just waiting for approval.

**Review by cticenhour — CHANGES_REQUESTED** [review]
Since Logan is out this week, I will take this over. Considering his comment about an appropriate issue number in an earlier commit and the fact that these changes were eventually reverted anyway, could you squash this into one commit? Please also change the title to be more descriptive of the changes, and note in the first description comment as to whether this closes #26493.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**
